lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,802 @@
+    return $t->success(is($t->tx->res->headers->location, $value, $desc));
+  };
+  $t->get_ok('/')
+    ->status_is(302)
+    ->$location_is('http://mojolicious.org')
+    ->or(sub { diag 'Must have been Joel!' });
+
+We create an anonymous subroutine (C<$location_is>) whose method signature
+matches other C<*_is> methods in L<Test::Mojo>. We create a default test
+description (C<$desc>), set the L<Test::Builder> C<Level> global variable one
+level higher (which tells L<Test::Builder> how far up the call stack to look
+when something fails). Finally we use L<Test::More>'s C<is> method to compare
+the location header with the expected header value. We wrap that in the
+C<success> method, which propagates the object for method chaining.
+
+=head3 Subclassing
+
+In some situations you may find your tests have a lot of boilerplate setup
+(authentication, etc.) and may want to subclass L<Test::Mojo> to handle
+that. Because of some of the expectations of L<Test::Mojo> and how it handles
+configuration data, subclassing takes a little extra work.
+
+For an example let's say we have an RPC-based web application I<that is
+already deployed>, where the HTTP method is always C<POST> and the remote
+procedure name is in the URL, and we have a C<Basic> authorization header with
+username C<me> and password C<secretly>.
+
+All of our requests follow this basic pattern:
+
+  POST /<service>/<method>
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"json":"payload"}
+
+For example:
+
+  POST /bedroom-light/power-on
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"level":"50%"}
+
+Or:
+
+  POST /blender/spin-blades
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"speed":"whip"}
+
+Or:
+
+  POST /thermostat/heat
+  Host: myapp.example.com
+  Authorization: Basic bWU6c2VjcmV0bHkK=
+
+  {"temperature":"23°C"}
+
+To test this kind of web application, we could do something like this for
+every test:
+
+  $t->post_ok('http://myapp.example.com/thermostat/heat',
+              {Authorization => 'Basic bWU6c2VjcmV0bHkK='},
+              json => {temperature => '23°C'})
+    ->status_is(200);
+
+But what we'd like to do is to make the tests as simple as regular
+L<Mojolicious> tests:
+
+  $t->post_ok('/thermostat/heat',
+              json => {temperature => '23°C'});
+
+To do this we're going to override L<Test::Mojo>'s constructor and L<post_ok>
+methods. The constructur will accept a C<base_url>, C<username>, and
+C<password> parameters. The C<post_ok> method will add the C<Authorization>
+header for us.
+
+Because L<Test::Mojo>'s constructor has some shortcuts built into it for
+passing configuration data to Mojolicious applications, we can't invoke
+C<SUPER::new> without removing our custom arguments first:
+
+  package Test::MyApp;
+  use Mojo::Base 'Test::Mojo';
+  use Mojo::Util 'b64_encode';
+
+  has [my @attr = qw/base_url username password/];
```

Comment:
For package scope variables i like to use upper case (`@ATTR`), not sure if that would be distracting here.